app/javascript/components/patient/follow_up_flag/FollowUpFlag.js

```diff
@@ -0,0 +1,278 @@
+    if (event.target.id == 'set_flag_for_follow_up') {
+      this.setState({ clear_flag: false });
+    } else if (event.target.id == 'clear_flag_for_follow_up') {
+      this.setState({ clear_flag: true });
+    } else {
+      this.setState({ [event.target.id]: value });
+    }
+  };
+
+  handleApplyHouseholdChange = apply_to_household => {
+    const no_members_selected = apply_to_household && this.state.apply_to_household_ids.length === 0;
+    this.setState({ apply_to_household, no_members_selected });
+  };
+
+  handleApplyHouseholdIdsChange = apply_to_household_ids => {
+    const no_members_selected = this.state.apply_to_household && apply_to_household_ids.length === 0;
+    this.setState({ apply_to_household_ids, no_members_selected });
+  };
+
+  /**
+   * Handles a key press event on the search form control.
+   * Checks for enter button press and prevents submisson event.
+   * @param {Object} event
+   */
+  handleKeyPress = event => {
+    if (event.which === 13) {
+      event.preventDefault();
+    }
+  };
+
+  // Makes a POST to update the follow-up flag for the current patient.
+  submit = () => {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+
+      // Different POST requests are used depending on whether this action is triggered via the dashboard or monitoree page
+      if (this.props.bulk_action) {
+        let idArray = this.props.patients.map(x => x['id']);
+        axios
+          .post(window.BASE_PATH + '/patients/bulk_edit', {
+            ids: idArray,
+            bulk_edit_type: 'follow-up',
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.bulk_action_apply_to_household,
+          })
+          .then(() => {
+            location.href = window.BASE_PATH;
+          })
+          .catch(error => {
+            reportError(error);
+            this.setState({ loading: false });
+          });
+      } else {
+        axios
+          .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/follow_up_flag', {
+            follow_up_reason: this.state.follow_up_reason,
+            follow_up_note: this.state.follow_up_note,
+            clear_flag: this.state.clear_flag,
+            clear_flag_reason: this.state.clear_flag_reason,
+            apply_to_household: this.state.apply_to_household,
+            apply_to_household_ids: this.state.apply_to_household_ids,
+          })
+          .then(() => {
+            // Reload the page to see the flag on the monitoree's page
+            location.reload();
+          })
+          .catch(err => {
+            reportError(err?.response?.data?.error ? err.response.data.error : err, false);
+          });
+      }
+    });
+  };
+
+  render() {
```

Comment:
Now that I played with this a bit, I don't love the radio buttons.  We don't use them for bulk action or for setting the flag, only when editing/removing which is still a hair confusing.  What are your thoughts with removing them and when a flag is set have a remove flag option? Then you could have the modal title update depending on what action is being taken

Reply:
I was more thinking removing the radio buttons all together and just have the modal dynamically show based on if you clicked the clear or set/edit button to load it.  Not totally sure how that would work for bulk actions but I think it would be worth asking the PO team about

For example here just adding like Edit / Delete and have two different hyperlinks or something

<img width="154" alt="Screen Shot 2021-05-28 at 10 08 01 AM" src="https://user-images.githubusercontent.com/35042815/119996372-ae007100-bf9c-11eb-840b-53f807ebb041.png">

Reply:
I see what you are trying to get at. If it weren't for the bulk action, I would agree with you. 

If we didn't include the edit/clear options in the same modal, the only way we'd be able to handle clearing a flag via bulk action is to have a completly separate "Clear Follow-up Flag" bulk action in the drop down. This strategy doesn't appeal to me though becuse it leads to the situation where you can select _only_ non-flagged monitorees and select that clear flag bulk action option which doesn't make a lot of sense. Unless of course we deactivate/hide that drop down option in that situation which doesn't sound ideal to me, but is a possibility. 

So given the circumstances, I still believe the radio buttons are a more streamlined solution in comparison to the seperat Edit/Clear Flag options. What do you think, still worth asking the PO team?

Reply:
I see what you're saying, I still think at least on an individual standpoint it would look better and we could keep it for bulk action.  Idk if there is some other way that would also make it look better? or hide the radio buttons instead of disabling the one when clearing isn't an option?

Reply:
@sgober and I talked offline about this - We decided to keep the radio buttons for the bulk action. When updating a flag via the monitoree's profile, we have removed the radio buttons. In this case, there will instead be edit/delete icon buttons in the Flag panel that the user can select to open the modal to the appropriate view. The modal's title and associacted submission button text reflects the specifics of these actions.

This has been addressed along with relevant test updates in my most recent commit.